### DISCUSSION TITLE
Logging Original and Proxied Requests

### DISCUSSION THREAD

**Discussion by app2eds** [dotnet/yarp#397] [General] [answered]
Looking for the easiest way to log incoming requests along with the proxied requests.  This would be primarily for debugging and troubleshooting to ensure that my reverse proxy routes and clusters settings are correct.  Thanks in advance.

**Comment by Tratcher** [comment, marked as answer]
What fields do you expect to be logged? ASP.NET Core already [logs](https://github.com/dotnet/aspnetcore/blob/1e7d6d56a5ae6b9ddfc8680d6f4f8282a8f17400/src/Hosting/Hosting/src/Internal/HostingRequestStartingLog.cs#L50) these for the incoming request:
- HTTP version
- Method
- Full uri
- Content-type
- Content-Length

And for [responses](https://github.com/dotnet/aspnetcore/blob/1e7d6d56a5ae6b9ddfc8680d6f4f8282a8f17400/src/Hosting/Hosting/src/Internal/HostingRequestFinishedLog.cs#L61):
- Status code
- Content-Length
- Content-Type
- Elapsed time

HttpProxy currently logs the absolute uri for the outgoing request.
https://github.com/microsoft/reverse-proxy/blob/43929dc9894ad3b986b61fe9966610297f0232e2/src/ReverseProxy/Service/Proxy/HttpProxy.cs#L370

**Reply by app2eds** [reply]
Thank you!  I had my logging level set to Warning so I wasn't seeing the logs.
